metrics/src/metrics.rs

```diff
@@ -88,25 +88,34 @@ impl Default for MetricsAgent {
         Self::new(
             Arc::new(InfluxDbMetricsWriter::new()),
             Duration::from_secs(10),
+            //max per-second datapoint submission limit
+            4000,
         )
     }
 }
 
 impl MetricsAgent {
-    fn new(writer: Arc<MetricsWriter + Send + Sync>, write_frequency: Duration) -> Self {
+    fn new(
+        writer: Arc<MetricsWriter + Send + Sync>,
+        write_frequency: Duration,
+        rate_limit: usize,
```

Comment:
Oh let's include units here: `rate_limit` -> `max_points_per_sec`

Reply:
Sure.  I don't know if this actually does need to be a usize, as its only useful after being multiplied by write_frequency, which will either end up as an unsigned int or a float.  The product is all that's needed as usize.